packages/roosterjs-editor-dom/lib/utils/setColor.ts

```diff
@@ -34,5 +51,100 @@ export default function setColor(
                 element.dataset[dataSetName] = modeIndependentColor.lightModeColor;
             }
         }
+
+        if (isBackgroundColor && shouldAdaptTheFontColor) {
+            adaptFontColorToBackgroundColor(element, isDarkMode);
+        }
     }
 }
+
+/**
+ * Change the font color to white or some other color, so the text can be visible with a darker background
+ * @param element The element that contains text.
+ */
+function adaptFontColorToBackgroundColor(element: HTMLElement, isDarkMode?: boolean) {
+    if (element.firstElementChild?.hasAttribute('style')) {
+        return;
+    }
+    const backgroundColor = element.style.getPropertyValue('background-color');
+    const lightModeBackgroundColor = isDarkMode
+        ? element.dataset[DarkModeDatasetNames.OriginalStyleBackgroundColor] ||
+          element.dataset[DarkModeDatasetNames.OriginalAttributeBackgroundColor]
+        : backgroundColor;
+    if (!lightModeBackgroundColor || lightModeBackgroundColor === TRANSPARENT) {
+        return;
+    }
+    const isADarkOrBrightOrNone = isADarkOrBrightColor(lightModeBackgroundColor!);
+    switch (isADarkOrBrightOrNone) {
+        case ColorTones.DARK:
+            const colorDark = isDarkMode ? lightModeBackgroundColor : WHITE;
+            element.dataset[DarkModeDatasetNames.OriginalStyleColor] = WHITE;
+            setColor(element, colorDark, false);
```

Comment:
This is not right.

now you already know whether you should apply white color for light mode. What you need is just set white color. You can define a mode independent white color as:
```ts
const WHITE: ModeIndependentColor = {
  lightColor: '#FFFFFF',
  darkColor: '#333333', // This is the dark color value for white in dark mode
};
```

Then use setColor here to set this color:
```ts
setColor(element, WHITE, false /*isBackground*/ isDarkMode);
```

Similar for the Light case.

Reply:
No need to manually set element.dataset[DarkModeDatasetNames.OriginalStyleColor] here because setColor will do that for you.